training/COMPARISON.md

```diff
@@ -0,0 +1,462 @@
+
+- [Pandas DataFrame Documentation](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html)
+- [Arkouda DataFrame Documentation](https://bears-r-us.github.io/arkouda/usage/dataframe.html?highlight=dataframe#arkouda.DataFrame)
+
+<a id="df_pd_create"></a>
+### Creating Pandas DataFrame
+Pandas can generate a DataFrame from data where each column is a Python List or a `NumPy` array. The dictionary structure is demonstrated below:
+
+```json
+{
+    "column_name_1": [0, 1, 2],
+    "column_name_2": ["a", "b", "c"]
+}
+```
+
+```python
+import pandas as pd
+pd_df = pd.DataFrame({
+    'F_Name': ['John', 'Jane', 'John', 'Jake'],
+    'L_Name': ['Doe', 'Doe', 'Smith', 'FromStateFarm'],
+    'Age': [37, 35, 50, 32],
+    'Salary': [75000, 77000, 100000, 35000]
+})
+pd_df
+  F_Name         L_Name  Age  Salary
+0   John            Doe   37   75000
+1   Jane            Doe   35   77000
+2   John          Smith   50  100000
+3   Jake  FromStateFarm   32   35000
+```
+
+<a id="df_ak_create"></a>
+### Creating Arkouda DataFrame
+Arkouda DataFrames require the columns to be `pdarray` objects. The dictionary structure is below:
+```json
+{
+   "column_name_1": ak.array([0, 1, 2]),
+   "column_name_2": ak.array(["a", "b", "c"])
+}
+```
+
+```python
+import arkouda as ak
+ak.connect()
+ak_df = ak.DataFrame({
+    'F_Name': ak.array(['John', 'Jane', 'John', 'Jake']),
+    'L_Name': ak.array(['Doe', 'Doe', 'Smith', 'FromStateFarm']),
+    'Age': ak.array([37, 35, 50, 32]),
+    'Salary': ak.array([75000, 77000, 100000, 35000])
+})
+ak_df
+  F_Name         L_Name  Age  Salary
+0   John            Doe   37   75000
+1   Jane            Doe   35   77000
+2   John          Smith   50  100000
+3   Jake  FromStateFarm   32   35000 (4 rows x 4 columns)
+```
+
+<a id="df_groupby"></a>
+### GroupBy on DataFrames
+`GroupBy` on DataFrames is similar in concept to `GroupBy` on arrays. When calling `.count()` on the resulting `GroupBy` object there is some variance in the return from Arkouda and `Pandas`. 
+
+#### GroupBy on Pandas DataFrame
+One important note is that Pandas support grouping on both axes. Arkouda does not currently support this.
+```python
+import pandas as pd
+pd_df = pd.DataFrame({
+    'F_Name': ['John', 'Jane', 'John', 'Jake'],
+    'L_Name': ['Doe', 'Doe', 'Smith', 'FromStateFarm'],
+    'Age': [37, 35, 50, 32],
+    'Salary': [75000, 77000, 100000, 35000]
+})
+pd_g = pd_df.groupby("F_Name")
+pd_g.count()
+        L_Name  Age  Salary
+F_Name                     
+Jake         1    1       1
+Jane         1    1       1
+John         2    2       2
+```
+Notice that the return of count is a `DataFrame` object with each value representing the number of different values in a given column for the corresponding grouped column.
+
+#### GroupBy on Arkouda DataFrame
+Running `GroupBy` on and Arkouda `DataFrame` results in an Arkouda `GroupBy` object. This results in different output when running `.count()` on the result.
```

Comment:
`on and Arkouda DataFrame` -> `on an Arkouda DataFrame`

Reply:
updated